pilot/pkg/model/sidecar.go

```diff
@@ -231,13 +231,67 @@ func convertIstioListenerToWrapper(ps *PushContext, configNamespace string,
 		ConfigNamespace: configNamespace,
 	}
 
-	out.services = out.selectServices(ps.Services(&dummyNode))
+	out.services = out.selectServices(ps.Services(&dummyNode), configNamespace)
 	meshGateway := map[string]bool{IstioMeshGateway: true}
 	out.virtualServices = out.selectVirtualServices(ps.VirtualServices(&dummyNode, meshGateway))
 
 	return out
 }
 
+// ServiceForHostname returns the service associated with a given hostname following SidecarScope
+func (sc *SidecarScope) ServiceForHostname(hostname Hostname, serviceByHostname map[Hostname]map[string]*Service) *Service {
+	if sc == nil {
+		// No Sidecar scope. An arbitrary service will be chosen
```

Comment:
Why arbitrary ? 

And do we still have the case of no sidecar scope ? I thought we create a dummy one by default.

Reply:
In the code we still do  have cases with no sidecar. I don't fully understand it thouugh.. We have this variable
```
	// edsClusters keep tracks of all watched clusters in 1.0 (not isolated) mode
	// TODO: if global isolation is enabled, don't update or use this.
	edsClusters = map[string]*EdsCluster{}
```

Yet we use it all over the place - most notably, AdsPushAll will use this and it does not have access to any specific proxy/sidecar scoping.

Specifically updateClusterInc will call this with no sidecar scope.


Should all of this logic be removed and just be using pushConnection instead of PushAll?